obspy/clients/fdsn/mass_downloader/__init__.py

```diff
@@ -0,0 +1,503 @@
+     time span.
+   * It can relaunch download to acquire missing pieces which might happen for
+     example if a data center has been offline.
+   * It can assure that there always is a corresponding StationXML file for the
+     waveforms.
+
+Usage Examples
+--------------
+
+Before delving into the nitty-gritty details of how it works and why it does
+things in a certain way we'll demonstrate the usage of this module on two
+annotated examples. They can serve as templates for your own needs.
+
+Earthquake Data
+~~~~~~~~~~~~~~~
+
+The classic seismological data set is consisting of waveform recordings for a
+certain earthquake. This example downloads all data it can find for the
+Tohoku-Oki Earthquake from 5 minutes before the earthquake centroid time to 1
+hour after.  It will furthermore only download data with an epicentral distance
+between 70.0 and 90.0 degrees and some additional restrictions. Be aware that
+this example will attempt to download data from all FDSN data centers that
+ObsPy knows of and combine it into one data set.
+
+.. code-block:: python
+
+    import obspy
+    from obspy.clients.fdsn.mass_downloader import CircularDomain, \\
+        Restrictions, MassDownloader
+
+    origin_time = obspy.UTCDateTime(2011, 3, 11, 5, 47, 32)
+
+    # Circular domain around the epicenter. This will download all data between
+    # 70 and 90 degrees distance from the epicenter. This module also offers
+    # rectangular and global domains. More complex domains can be defined by
+    # inheriting from the Domain class.
+    domain = CircularDomain(latitude=37.52, longitude=143.04,
+                            minradius=70.0, maxradius=90.0)
+
+    restrictions = Restrictions(
+        # Get data from 5 minutes before the event to one hours after the
+        # event. This defines the temporal bounds of the waveform data.
+        starttime=origin_time - 5 * 60,
+        endtime=origin_time + 3600,
+        # You might not want to deal with gaps in the data. If this settings is
+        # True, any trace with a gap/overlap will be discarded.
+        reject_channels_with_gaps=True,
+        # And you might only want waveforms that have data for at least 95 % of
+        # the requested time span. Any trace that is shorter than 95 % of the
+        # desired total duration will be discarded.
+        minimum_length=0.95,
+        # No two stations should be closer than 10 km to each other. This is
+        # useful to for example filter out stations that are part of different
+        # networks but at the same physical station. Settings this to zero or
+        # None will disable that filtering.
+        minimum_interstation_distance_in_m=10E3,
+        # Only HH or BH channels. If a station has HH channels, those will be
+        # downloaded, otherwise the BH. Nothing will be downloaded if it has
+        # neither. You can add more/less patterns if you like.
+        channel_priorities=("HH[Z,N,E]", "BH[Z,N,E]"),
```

Comment:
`bash` would accept `{Z,N,E}` which might be what you're thinking of, but I think it's an extension to normal POSIX syntax and `fnmatch` doesn't accept it.